### DISCUSSION TITLE
Where or how can I download older versions of GitHub Copilot and GitHub Copilot Chat VSIX?

### DISCUSSION THREAD

**Discussion by ciphero** [community/community#139182] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

There are only ever five VSIX downloadable items listed on the version history page, ex:
https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat&ssr=false#version-history

Version | Last Updated |  
-- | -- | --
0.21.2024092002 | an hour ago | Download
0.21.2024091903 | 21 hours ago | Download
0.21.2024091902 | Thursday | Download
0.21.2024091804 | Wednesday | Download
0.21.2024091803 | Wednesday | Download

**Comment by sam-math** [comment]
Try this: [https://code.visualstudio.com/updates/v1_30#_install-previous-versions](https://code.visualstudio.com/updates/v1_30#_install-previous-versions)

If you're interested in the vsix file and know the specific version, just replace the version part on the URL: [https://marketplace.visualstudio.com/_apis/public/gallery/publishers/GitHub/vsextensions/copilot-chat/_0.21.2024091803_/vspackage](https://marketplace.visualstudio.com/_apis/public/gallery/publishers/GitHub/vsextensions/copilot-chat/0.21.2024091803/vspackage)

**Reply by ciphero** [reply]
Thank you for the suggestions. Unfortunately I am using Codium (VSCodium) so maybe the features work a little differently but when I tried the _install-previous-version_ option it is greyed out on the manually installed VSIX packages I have.
Unfortunately I don't know what all the older versions are named, otherwise the second solution might have worked, given the older VSIX packages are still available for download. I am not sure if there is a list of all previously released packages, if so, I could give that a try.
For ex I'm looking for all the VSIX packages for GitHub Copilot Chat that are supported with VSCode/Codium version 1.93.x.

**Comment by ciphero** [comment, marked as answer]
I managed to resolve the issue using a three-step solution to search extension release history and then download and install the specific version as suggested by @sam-math.

1. vsce show GitHub.copilot-chat --json | jq   # (and find a compatible older release version, e.g. 0.20.2024091202)
2. wget https://marketplace.visualstudio.com/_apis/public/gallery/publishers/GitHub/vsextensions/copilot-chat/0.20.2024091202/vspackage
3. manually install VSIX in Codium

Hope it is helpful for more people.

**Reply by deemeetree** [reply]
The best answer! Works like a charm for using it with other VSCode based tools like Windsurf AI :)

**Comment by Drek282** [comment]
@ciphero

To add some missing instructions for nubs such as myself.

1. If you don't have this already, you will need to install the npm package ("sudo apt install npm" on Debian).
2. Then you will need to install vsce ("sudo npm install vsce -g"—the " -g" flag will install vsce globally).
3. Then follow the above instructions.
4. Extract the resulting "vspackage" twice.  cd to "extension" directory.
5. run "vsce package" in that directory and you will get your vsix copilot-chat extension.

**NOTE**:  The most recent version of Copilot Chat that I was able to install was 0.23.2024121102.  That is one major version, 0.23.x, behind the most current release, as of this post, which is 0.24.x.

**Comment by skoonin** [comment]
I was able to get it installed, up to 0.22.2024100702 Only strange thing is I don't have access to claude, just the gpt's.  Anyone have a solution for this?

**Comment by VGerris** [comment]
https://github.com/VSCodium/vscodium/discussions/1487#discussioncomment-13927656 check there for newer versions, which you need for new model support

**Comment by ablof** [comment]
The VSIX of older versions are also available here: https://github.com/microsoft/vscode-copilot-chat/releases
